src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransportFactory.php

```diff
@@ -42,4 +43,17 @@ protected function getSupportedSchemes(): array
     {
         return ['telegram'];
     }
+
+    private function getToken(Dsn $dsn): string
+    {
+        if (null === $dsn->getUser() && null === $dsn->getPassword()) {
+            throw new IncompleteDsnException('Missing token');
```

Comment:
I haven't tested this and I don't know this well ... but I wonder if we could update this error message to be more useful. Instead of `Missing token`, we could say something like `The given DSN string is missing either the user, the password or both.`

Reply:
For the Telegram transport there is no username nor password, but only a token that has to be configured. It's just because of how the DSN is parsed that the actual token value will be split into the username and password because of the colon that is part of the token.